### PR TITLE
<carry>: enable CSI migration gates in Attach/Detach controller

### PR DESCRIPTION
**Pull request by jsafrane**
Run A/D controller with CSI migration feature flags force-enabled.

There are two commits.:
1. `UPSTREAM: 99942` is already in Kubernetes 1.21 and we'll get it through rebase.
2. very minimal `<carry>` patch that forcefully sets CSI migration gates *only* in A/D controller. We will need it as `<carry>` patch until CSI migration of all volume plugins reaches GA (1.23-1.24?)

See https://github.com/openshift/enhancements/pull/549 for details.

The patch works around two issues we have in OCP:

* CVO downgrades Kubernetes components in wrong order: kube-controller-manager (A/D controller) is downgraded first and then it downgrades kubelets. This is not supported Kubernetes version skew, kubelet cannot be newer than KCM. A/D controller breaks in this case.

* FeatureGates are applied in random order. Without this PR, we need to enable CSI migration flag in A/D controller first, then kubelets. With this PR, the order does not matter.

With this PR, A/D controller knows about CSI migration and will attach volumes to nodes either "in-tree style" or "CSI style", depending what the node actually wants. In a default 4.8 cluster, it will be in-tree style, as CSI migration is disabled everywhere else, so there should be no visible change in A/D controller processing. It becomes useful as the CSI migration becomes enabled on nodes (either during upgrade to 4.9 or downgrade 4.9->4.8), or when user opts-in for CSI migration as tech preview in 4.8 via FeatureGates CR.

### CODE REVIEW

File: pkg/controller/volume/attachdetach/openshift_patch.go
Comment (sttts) [inline pkg/controller/volume/attachdetach/openshift_patch.go:9]: patchADCFeatureGates suggests that this is has side-effect. Instead call it `patchedADCFeatureGates` or `customizedADCFatureGates`.

**Review by soltysh — APPROVED** [review]
Please provide more information to the carry commit, pointing to https://github.com/openshift/enhancements/pull/549 and that this is owned by storage team and should be contacted with them around 3-4 releases, so around 4.11/4.12 for a potential drop. 

The change lgtm, but I'd like architects to sign off the enhancement first before merging this.

**Comment by soltysh** [comment]
> The change lgtm, but I'd like architects to sign off the enhancement first before merging this.

/hold

**Comment by jsafrane** [comment]
@soltysh better comment added + linked the enhancement in the commit message.

**Comment by jsafrane** [comment]
Rebased to 1.21

**Review by bertinatto — COMMENTED** [review]
/lgtm

**Comment by bertinatto** [comment]
/test e2e-aws-serial

```
=Error: error updating LB Target Group (arn:aws:elasticloadbalancing:us-east-1:460538899914:targetgroup/ci-op-igp5p5rx-12284-ffrf5-sint/54dd658f69441292) tags: error tagging resource (arn:aws:elasticloadbalancing:us-east-1:460538899914:targetgroup/ci-op-igp5p5rx-12284-ffrf5-sint/54dd658f69441292): TargetGroupNotFound: Target groups 'arn:aws:elasticloadbalancing:us-east-1:460538899914:targetgroup/ci-op-igp5p5rx-12284-ffrf5-sint/54dd658f69441292' not found
```

**Review by soltysh — APPROVED** [review]
/approve

**Pull request merged**
